safe/gui/tools/osm_downloader_dialog.py

```diff
@@ -344,6 +351,79 @@ def accept(self):
             # Unlock the groupbox
             self.groupBox.setEnabled(True)
 
+    def get_output_base_path(
+            self,
+            output_directory,
+            output_prefix,
+            feature_type,
+            overwrite):
+        """Get a full base name path to save the shapefile.
+
+        :param output_directory: The directory where to put results.
+        :type output_directory: str
+
+        :param output_prefix: The prefix to add for the shapefile.
+        :type output_prefix: str
+
+        :param feature_type: What kind of features should be downloaded.
+            Currently 'buildings', 'building-points' or 'roads' are supported.
+        :type feature_type: str
+
+        :param overwrite: Boolean to know if we can overwrite existing files.
+        :type overwrite: str
+
+        :return: The base path.
+        :rtype: str
+        """
+        path = os.path.join(
+            output_directory, '%s%s' % (output_prefix, feature_type))
+
+        if not overwrite:
+            separator = '-'
+            suffix = self.get_unique_suffix_file_path(
+                '%s.shp' % path, separator)
+
+            if suffix:
+                path = os.path.join(output_directory, '%s%s%s%s' % (
+                    output_prefix, feature_type, separator, suffix))
+
+        return path
+
+    @staticmethod
+    def get_unique_suffix_file_path(file_path, separator='-', i=0):
+        """Check and return the minimum number to suffix the file with a
```

Comment:
Docstrings should be one precis sentence followed by a full stop, line break and then additional detail.